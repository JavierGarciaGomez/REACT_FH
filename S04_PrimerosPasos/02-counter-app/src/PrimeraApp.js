// 35, 36, 37, 38, 39, 40

import { Fragment } from "react";
// 39
import PropTypes from "prop-types";

// Functional component
// const PrimeraApp = (props) => {
const PrimeraApp = ({ saludo, defaultMsj }) => {
  const variableString = "texto en variable";

  return (
    <Fragment>
      <h1> Mi {variableString}</h1>
      <p>un párrafo con {saludo}.</p>
      <p>40, esto es un mensaje por defecto: {defaultMsj}.</p>
    </Fragment>
  );
};

// 39 validación de proptypes
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  defaultMsj: "Mensaje por default",
};
export default PrimeraApp;
