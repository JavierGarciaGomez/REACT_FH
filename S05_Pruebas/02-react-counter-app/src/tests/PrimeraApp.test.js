// 58, 59
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import React, { Fragment } from "react";
import { shallow } from "enzyme";

describe("pruebas en <Primera App>", () => {
  test("debe retornar el saludo ", async () => {
    const saludo = "Hola, soy anónimo";
    const wrapper = render(<PrimeraApp saludo={"anónimo"} />);

    expect(wrapper.getByText(saludo)).toBeInTheDocument();
  });

  //   59 Pruebas enzyme
  test("debe mostrar <PrimeraApp> ", async () => {
    const saludo = "Hola, soy Juan";
    const wrapper = shallow(<PrimeraApp saludo={"Juan"} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtítulo ", async () => {
    const saludo = "Hola, soy Juan";
    const subtitulo = "Soy un subtítulo!";
    const wrapper = shallow(
      <PrimeraApp saludo={"Juan"} subtitulo={subtitulo} />
    );

    const textoParrafo = wrapper.find("p").text();
    console.log(textoParrafo);

    expect(textoParrafo).toBe(subtitulo);
  });
});
