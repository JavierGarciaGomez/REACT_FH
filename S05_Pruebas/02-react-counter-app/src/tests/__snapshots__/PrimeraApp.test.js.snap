// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Primera App> debe mostrar <PrimeraApp>  1`] = `
<Fragment>
  <h1>
     Hola, soy 
    Juan
     
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
